components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCPermissionBasedInternalScopeValidator.java

```diff
@@ -189,11 +194,22 @@ Note that if there is association between a federated user and local user () 'As
              */
             if (authenticatedUser.isFederatedUser()) {
                 /*
-                There is a flow where 'Assert identity using mapped local subject identifier' flag enabled but the
-                federated user doesn't have any association in localIDP, to handle this case we check for 'Assert
-                identity using mapped local subject identifier' flag and get roles from userStore.
-                 */
-                if (isSPAlwaysSendMappedLocalSubjectId(clientId)) {
+                If the role-based authorization feature is enabled & particular applications in the listed under the
+                required application list then retrieve permissions from the FIdp user roles.
+                Since tenant flow is enabled, permission will be fetched from user login(federated) tenant
+                not the application tenant.
+                */
+                if (IdentityUtil.getPropertyAsList(FIDP_ROLE_BASED_AUTHZ_APP_CONFIG)
+                        .contains((OAuth2Util.getServiceProvider(clientId).getApplicationName()))) {
+                    allowedUIResourcesForUser =
```

Comment:
Can we change the variable name to not have "UI" as it is incorrect

Reply:
Fixed with 462d62780a3b8d2046bf6f5b155f3e0813c0cc5e